src/box/journal.h

```diff
@@ -291,6 +273,17 @@ journal_write_try_async(struct journal_entry *entry)
 	return current_journal->write_async(current_journal, entry);
 }
 
+/** Write a single entry to the journal in synchronous way. */
+static inline int
+journal_write(struct journal_entry *entry)
+{
+	if (journal_write_try_async(entry) != 0)
```

Comment:
Not really. Previously it would be probably more reactive, because any blocking commit would wakeup all the pending txns and then go in too. Now the txn goes to the end of the queue and just waits until the current WAL write is done.

The only theoretical change is that now the WAL batching might improve (blocking txns will also stand into the queue without rushing into the WAL thread), while reactivity might decrease.